jahsieAult/homework2Answers.md

```diff
@@ -0,0 +1,308 @@
+You can also print your return value with an expression like `console.log(fractionString(7,4))`.
+
+**b)** Write a second version of your function which improves the output in two special cases:
+
+- A output string like "0 1/2" should be simplified to just "1/2";
+- A output string like "1 0/3" should be simplified to just "1".
+
+```
+function fractionString(n,d) {
+	var integer = (Math.floor(n/d)).toString(),
+    numerator = (n % d).toString(),
+    denominator = d.toString();
+    if (integer == 0) {
+    	finalValue = numerator + "/" + denominator;
+	}
+	else if (numerator == 0) {
+		finalValue = integer;
+	}
+	else {
+		finalValue = integer + " " + numerator + "/" + denominator;
+	}
+	return finalValue;
+}
+
+```
+
+---
+
+**2)** _(Moderate, 15%)_
+
+Rewrite each block below as the simplest equivalent you can discover.  Your equivalent should produce the same final conditions as the original code given the same initial conditions.
+In each case, if there is an undeclared variable, assume it has been declared earlier and set to an unknown value.
+
+**a)**
+
+```
+var i;
+if ((n - Math.floor(n))) >= .5) {
+   i = Math.ceil(n);
+} else {
+  i = Math.floor(n);
+}
+```
+Answer:
+
+```
+var i;
+i = Math.round(n);
+```
+
+**b)**
+
+```
+var y, xIsFalse = (x? false : true);
+if (xIsFalse)
+   y = false;
+else
+   y = x;
+```
+Answer:
+
+```
+var xIsFalse = true;
+if (xIsFalse) y = false;
+
+```
+
+**c)**
+
+```
+for (var count = 15 - i ; count < 15 ; count=count+1) {
+    i = i-1;
+    console.log(i+1)
+}
+```
+Answer:
+
+```
+for (var i; i > 0 ; i--) {
+    console.log(i);
+}
+
+```
+
+**d)**
+
+```
+var x;
+if (a) {
+   if (b) {
+     x = 0;
+   } else {
+     x = 1;
+   }
+} else {
+  if (b) {
+    x = 1;
+  } else {
+    x = 2;
+  }
+}
+```
+Answer:
+
+```
+var x;
+if (a && b) x = 0;
+	else if (a || b) x = 1;
+	else x = 2;
+```
+
+---
+
+**3)** _(Moderate, 20%)_
+
+Write a function that generates the lyrics to the Twelve Days of Christmas
+
+```
+function writeSong() {
+ 
+var startVerse = " day of Christmas\nmy true love sent to me:\n";
+
+var whatDay = ["first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"];
+var gift = ['Partridge in a Pear Tree\n', "Two Turtle Doves\n", "Three French Hens\n", "Four Calling Birds\n", "Five Golden Rings\n", "Six Geese a Laying\n", "Seven Swans a Swimming\n", "Eight Maids a Milking\n", "Nine Ladies Dancing\n", "Ten Lords a Leaping\n",  "Eleven Pipers Piping\n", "Twelve Drummers Drumming\n"];
+
+var song;
+var newGift = [];
+
+
+for (i = 0; i < whatDay.length; i++) { //creates loop
+  if (i < 1) {
+    song = "On the " + whatDay[i] + startVerse + "A " + gift[i] + "\n"
+  }
+  else {
+    newGift.unshift(gift[i]);
+    song += "On the " + whatDay[i] + startVerse + newGift.join("") + "And a " + gift[0] + "\n"; 
+  }
+}
+console.log(song);
+}
+writeSong()
```

Comment:
3) Nice! Way to go with integrating the "And a " into the code--when I did it, I had mine hardcoded into the partridge, which is more of a hack.